waku/v2/node/waku_payload.nim

```diff
@@ -77,3 +78,36 @@ proc encode*(payload: Payload, version: uint32, rng: var BrHmacDrbgContext):
       return err("Couldn't encode the payload")
   else:
     return err("Unsupported WakuMessage version")
+
+
+# Decodes a WakuMessage to a PayloadV2
+# Currently, this is just a wrapper over deserializePayloadV2 and encryption/decryption is done on top (no KeyInfo)
+proc decodePayloadV2*(message: WakuMessage): WakuResult[PayloadV2] 
```

Comment:
As an interim step I think having specific `decode` and `encode` functions for version `2` is ok, but we must merge this logic with [the existing `decodePayload` and `encode` procs](https://github.com/status-im/nim-waku/blob/430f2ecd6720d5835c430e310ffb06ce27f840c7/waku/v2/node/waku_payload.nim#L39-L80) if possible. I think that can be done by extending `KeyInfo` and having the return type of `decodePayload` be a case object rather than the version 1 specific `DecodedPayload`.

Reply:
I agree that this can be a separate PR. Thanks!

Reply:
To keep track of this, I created the following issue https://github.com/status-im/nim-waku/issues/942